File: WPFApplication3/Class1.pas
namespace 
WPFApplication3;

interface

type
  Class1 = public class
  private
  protected
  public
		class method Test(): System.Threading.Tasks.Task;
  end;

implementation

class method Class1.Test: System.Threading.Tasks.Task;
begin
  try
    raise new Exception('Parameter cannot be null');
  except
    var dial := new ClassLibrary1.TestDialogCL();
    await MaterialDesignThemes.Wpf.DialogHost.Show(dial, 'RootDialog');
  end;
end;

end.

File: ClassLibrary1/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassLibrary1
{
    public class Class1
    {
        public static async Task Test()
        {
            try
            {
                throw new System.Exception("Parameter cannot be null");
            }
            catch
            {
                var dial = new TestDialogCL();
                await MaterialDesignThemes.Wpf.DialogHost.Show(dial, "RootDialog");

            }
        }
    }
}
